vendor/assets/javascripts/initializers.js

```diff
@@ -54,4 +54,23 @@ $(document).on('turbolinks:load', function() {
       }
     });
   });
+  $('.js-dropzone-featured-image').each(function() {
+    $(this).find('form').dropzone({
+      dictDefaulMessage: 'Drop files or click here to upload',
+      init: function() {
+        this.on('success', function(req, res) {
+          var clone = $($.clone($('.js-featured-image-table-element')[0]));
+          var baseId = clone.find('input').attr('id').split('_');
+          baseId.pop();
+          baseId.push(res.id);
+          var radioButton = clone.find('input');
+          radioButton.attr('id', baseId);
+          radioButton.val(res.id);
+          radioButton.attr('checked', true);
+          clone.insertBefore($('.js-featured-image-table-element')[0]);
+          var featuredImage = $('.js-featured-image').attr('src', res.attachment.url);
```

Comment:
I like it. Like you said, needs some clean-up, better self-documenting structure. Also, we'll eventually need to be able to use this same function across a wide-range of asset upload situations, so specifics about the HTML classes and structure will need to be factored out somehow.